components/kyma-environment-broker/internal/process/upgrade_kyma/initialisation.go

```diff
@@ -59,6 +64,37 @@ func (s *InitialisationStep) Name() string {
 }
 
 func (s *InitialisationStep) Run(operation internal.UpgradeKymaOperation, log logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	o, err := s.orchestrationStorage.GetByID(operation.OrchestrationID)
+	if err != nil {
+		if dberr.IsNotFound(err) {
+			log.Errorf("orchestration %s not found: %v", err)
+			return s.operationManager.OperationFailed(operation, fmt.Sprintf("orchestration %s not found", operation.OrchestrationID))
+		}
+		log.Errorf("while getting orchestration from storage: %v", err)
+		return operation, s.timeSchedule.Retry, nil
+	}
+	if o.State == orchestrationExt.Canceled {
```

Comment:
Why do we need to check canceled orchestration state if canceled state is propagated down to operations?